flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamNode.java

```diff
@@ -265,17 +266,19 @@ public void setOperatorDescription(String operatorDescription) {
 
     public void setSerializersIn(TypeSerializer<?>... typeSerializersIn) {
         checkArgument(typeSerializersIn.length > 0);
-        this.typeSerializersIn = typeSerializersIn;
+        // Unfortunately code above assumes type serializer can be null, while users of for example
+        // getTypeSerializersIn would be confused by returning an array size of two with all
+        // elements set to null...
+        this.typeSerializersIn =
+                Arrays.stream(typeSerializersIn)
+                        .filter(typeSerializer -> typeSerializer != null)
+                        .toArray(TypeSerializer<?>[]::new);
     }
 
     public TypeSerializer<?>[] getTypeSerializersIn() {
```

Comment:
Does the code calling this method make no assumptions about the indices?
Let's say the original list of serializers contained 2 elements, with the first one being nul
l.
How does the code accessing this array figure out that what is now at index 0 is what it requires?

Reply:
It must be kept in sync with `StreamNode#getInEdges`. Each edge has it's own input serializer. It can never happen that first element is null, second is not null. Sources have zero inputs, one input task has single input, two input tasks have two inputs, multi input tasks have N inputs.

Reply:
Can this change result in the array being empty? If so, can this conflict with `StreamGraph#setSerializersFrom`?

Reply:
`StreamGraph#setSerializersFrom` was not used so I removed it in this PR :)

Yes the result can be empty for source tasks.